core/libretro-common/include/glsm/glsm.h

```diff
@@ -32,6 +32,7 @@
 RETRO_BEGIN_DECLS
 
 #ifdef HAVE_OPENGLES2
+typedef GLfloat GLdouble;
```

Comment:
this file is part of libretro-common, as such you can't change it that way since your change will be removed next time libretro-common is bumped.

Reply:
if you need custom declaration in glsm, i recommend using the glsym_private stuff i use in my yabasanshiro/kronos cores : https://github.com/libretro/yabause/tree/yabasanshiro/yabause/src/libretro
You need 3 things : 
- a header named `glsym_private.h` (see example above)
- a file with your custom functions (optional, see `glsym_private.c`, but i don't think you need it here)
- adding `-DHAVE_GLSYM_PRIVATE` to flags so that the glsm stuff will include the `glsym_private.h` header (however, i think this repo's libretro-common needs to be updated first)

Reply:
This line with `typedef GLfloat GLdouble` has been removed for whatever reasons here https://github.com/libretro/flycast/commit/3dc5a70a424e8e334ab64653d79e7221dc25d2a1#diff-814712f780385128ebce938849ace55eL35 & is normally part of libretro-common https://github.com/libretro/libretro-common/blob/master/include/glsm/glsm.h#L35 so basically he just restores what was removed before & breaks the build.

Reply:
this typedef was removed as the definition is in android ndk, but it is required for other openGL ES platform, so it should not have been removed in the first place as explained by 5shatten

Reply:
Ok, i checked latest libretro-common and it seems this def is in there, so there will be no problem with bumping it.

Reply:
Hi,
I need to remove `typedef GLfloat GLdouble; `on my Rockchip build

https://github.com/LibreELEC/libmali/blob/master/include/GLES2/gl2ext.h
have `typedef double GLdouble;`

/toolchain/armv8a-libreelec-linux-gnueabi/sysroot/usr/include/GLES2/gl2ext.h:3050:16: note: previous declaration as 'typedef double GLdouble'
 typedef double GLdouble;
                ^~~~~~~~
In file included from core/rend/gles/gles.h:7,


In addition, i have the same error on rpi4 build.

Reply:
opengl es 2 spec does not define GLdouble (cf page 12 of
https://www.khronos.org/registry/OpenGL/specs/es/2.0/es_full_spec_2.0.pdf and https://www.khronos.org/files/opengles20-reference-card.pdf)
that being said GLdouble are exposed through some extension , and the official khronos header do have the typedef
GLdouble is defined in openGL ES 3 spec

unfortunately the mesa gles2 dev lib installed on raspbian, and on buster on my PC, do not come with that definition
if you use a cross compilation toolchain, or the mali headers as it seems from your link you do have those...
not sure how this should be solved then as not all openGL ES headers will be the same :S

Reply:
https://github.com/libretro/libretro-common/issues/98

Reply:
ok, let me auto correct myself, GLdouble are defined on some platform in gl2ext.h as extension (thus the ext in the header file name)
they are not part of official/standard spec
so they are not on my buildchain, they are on yours :S
that doesn't help as to how to solve this...

Reply:
Hi again.
5schatten give us a proper solution :)
I have applied this one on mupen64plus too.

Reply:
Hi, I needed to remove `typedef GLfloat GLdouble;` to be able to compile it on my gameshell (allwinner based), so I think this modification should be done only if someone is trying to build for rpi4, not take as a general case, I don't know.